app/src/main/java/edu/mit/mitmobile2/RetrofitManager.java

```diff
@@ -0,0 +1,151 @@
+package edu.mit.mitmobile2;
+
+import java.lang.reflect.Field;
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import edu.mit.mitmobile2.shuttles.model.MITShuttlePredictionWrapper;
+import edu.mit.mitmobile2.shuttles.model.MITShuttleRouteWrapper;
+import edu.mit.mitmobile2.shuttles.model.MITShuttleStopWrapper;
+import edu.mit.mitmobile2.shuttles.model.MITShuttleVehiclesWrapper;
+import retrofit.Callback;
+import retrofit.Endpoint;
+import retrofit.RequestInterceptor;
+import retrofit.RestAdapter;
+import retrofit.http.GET;
+import timber.log.Timber;
+
+public class RetrofitManager {
+
+    private static class MitEndpoint implements Endpoint {
+
+        public static MitEndpoint create(String endpoint) {
+            return new MitEndpoint(endpoint);
+        }
+
+        public MitEndpoint(String url) {
+            this.url = url;
+        }
+
+        private String url;
+
+        public void setUrl(String url) {
+            this.url = url;
+        }
+
+        @Override
+        public String getUrl() {
+            if (url == null) {
+                throw new IllegalStateException("Url not set");
+            }
+            return url;
+        }
+
+        @Override
+        public String getName() {
+            return "default";
+        }
+    }
+
+    public static final String PROD_ENDPOINT = "http://m.mit.edu/apis";
```

Comment:
Good point! I didn't realize that the resources endpoint was different than all the others. I'll add some logic to grab the correct URL set from JSON based on the module that was passed in.